terraso_backend/apps/story_map/views.py

```diff
@@ -0,0 +1,213 @@
+            )
+            error_message = ErrorMessage(
+                code="update", context=ErrorContext(model="StoryMap", field=NON_FIELD_ERRORS)
+            )
+            return JsonResponse({"errors": [{"message": [asdict(error_message)]}]}, status=400)
+
+        story_map.title = form_data["title"]
+
+        if form_data["is_published"] == "true" and not story_map.is_published:
+            story_map.published_at = timezone.make_aware(
+                datetime.now(), timezone.get_current_timezone()
+            )
+
+        story_map.is_published = form_data["is_published"] == "true"
+
+        new_config = json.loads(form_data["configuration"])
+
+        story_map.configuration = handle_config_media(new_config, story_map.configuration, request)
+
+        entry_form = StoryMapForm(data=story_map.to_dict())
+
+        if not entry_form.is_valid():
+            error_messages = get_error_messages(entry_form.errors.as_data())
+            return JsonResponse(
+                {"errors": [{"message": [asdict(e) for e in error_messages]}]}, status=400
+            )
+
+        try:
+            story_map.save()
+        except IntegrityError as exc:
+            return handle_integrity_error(exc)
+
+        return JsonResponse(story_map.to_dict(), status=201)
+
+
+def handle_config_media(new_config, current_config, request):
+    for chapter in new_config["chapters"]:
+        media = chapter.get("media")
+        if media and "contentId" in media:
+            file_id = media["contentId"]
+            matching_file = next(
+                (file for file in request.FILES.getlist("files") if file.name == file_id), None
+            )
+            if matching_file:
+                url = story_map_media_upload_service.upload_file_get_path(
+                    str(request.user.id),
+                    matching_file,
+                    file_name=uuid.uuid4(),
+                )
+                chapter["media"] = {"url": url, "type": media["type"]}
+
+    if (current_config is None) or (not current_config.get("chapters")):
+        return new_config
+
+    # Delete changed media
+    current_media = [
+        chapter["media"]["url"]
+        for chapter in current_config["chapters"]
+        if chapter.get("media") and "url" in chapter["media"]
+    ]
+    new_media = [
+        chapter["media"]["url"]
+        for chapter in new_config["chapters"]
+        if chapter.get("media") and "url" in chapter["media"]
+    ]
+
+    for media_path in current_media:
+        if media_path not in new_media:
+            try:
+                story_map_media_upload_service.delete_file(media_path)
+            except Exception as e:
+                logger.exception(
+                    "Error deleting media file",
+                    extra={"media_path": media_path, "error": str(e)},
+                )
+
+    return new_config
+
+
+def handle_integrity_error(exc):
+    logger.info(
```

Comment:
question: what causes this error?

Reply:
@paulschreiber Right now this is triggered for the unique title validation